elastalert/alerts.py

```diff
@@ -219,7 +217,12 @@ def alert(self, matches):
 
         try:
             self.smtp = SMTP(self.smtp_host)
+            if(self.smtp_port):
+                self.smtp = SMTP(self.smtp_host,self.smtp_port)
+                self.smtp.ehlo()
+                self.smtp.starttls()
```

Comment:
This will throw an exception if starttls is not supported by the server. Specifying the smtp_port should not mean always use starttls. ehlo() will tell you if STARTTLS is supported, but it is not checked here.